libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -144,12 +144,16 @@ vector<ASTString> DeclarationContainer::similarNames(ASTString const& _name) con
 	for (auto const& declaration: m_declarations)
 	{
 		string const& declarationName = declaration.first;
+		if (_name.size() * declarationName.size() > 6400)
+			continue;
 		if (stringWithinDistance(_name, declarationName, MAXIMUM_EDIT_DISTANCE))
 			similar.push_back(declarationName);
 	}
 	for (auto const& declaration: m_invisibleDeclarations)
 	{
 		string const& declarationName = declaration.first;
+		if (_name.size() * declarationName.size() > 6400)
```

Comment:
Can you please move these checks into `stringWithinDistance`?

Reply:
This may lead to misused `stringWithinDistance` in the future, should we do that?

Reply:
You can add another parameter, just like `MAXIMUM_EDIT_DISTANCE`.